crates/router/src/db/merchant_account.rs

```diff
@@ -403,12 +403,23 @@ async fn publish_and_redact_merchant_account_cache(
         .publishable_key
         .as_ref()
         .map(|publishable_key| CacheKind::Accounts(publishable_key.into()));
+    let kgraph_key = merchant_account.default_profile.as_ref().map(|profile_id| {
+        CacheKind::KGraph(
+            format!(
+                "kgraph_{}_{}",
+                merchant_account.merchant_id.clone(),
+                profile_id,
+            )
+            .into(),
+        )
+    });
```

Comment:
We don't want to redact the cache only for the default profile. Again requires feature gating for the `business_profile_routing` feature.